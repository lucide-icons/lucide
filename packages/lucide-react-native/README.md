<p align="center">
  <a href="https://github.com/lucide-icons/lucide#gh-light-mode-only">
    <img src="https://lucide.dev/package-logos/lucide-react-native.svg#gh-light-mode-only" alt="Lucide React Native - Implementation of the lucide icon library for React Native applications." width="540">
  </a>
  <a href="https://github.com/lucide-icons/lucide#gh-dark-mode-only">
    <img src="https://lucide.dev/package-logos/package-logos/dark/lucide-react-native.svg#gh-dark-mode-only" alt="Lucide React Native - Implementation of the lucide icon library for React Native applications." width="540">
  </a>
</p>


# Lucide React Native

Implementation of the lucide icon library for React Native applications.

> What is lucide? Read it [here](https://github.com/lucide-icons/lucide#what-is-lucide).

## Installation

First, ensure that you have `react-native-svg` (version between 12 and 15) installed. Then, install the package:

```sh
yarn add lucide-react-native
```

or

```sh
npm install lucide-react-native
```

## Documentation

For full documentation, visit [lucide.dev](https://lucide.dev/guide/packages/lucide-react-native)

## Community

Join the [Discord server](https://discord.gg/EH6nSts) to chat with the maintainers and other users.

## License

Lucide is licensed under the ICS license. See [LICENSE](https://lucide.dev/license).

## Sponsors

<a href="https://vercel.com?utm_source=lucide&utm_campaign=oss">
  <img src="https://lucide.dev/vercel.svg" alt="Powered by Vercel" width="200" />
</a>

<a href="https://www.digitalocean.com/?refcode=b0877a2caebd&utm_campaign=Referral_Invite&utm_medium=Referral_Program&utm_source=badge"><img src="https://lucide.dev/digitalocean.svg" width="200" alt="DigitalOcean Referral Badge" /></a>

### Awesome backer üç∫

<a href="https://www.scipress.io?utm_source=lucide"><img src="https://lucide.dev/sponsors/scipress.svg" width="180" alt="Scipress sponsor badge" /></a>
