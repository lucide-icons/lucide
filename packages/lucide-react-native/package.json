{
  "name": "lucide-react-native",
  "description": "A Lucide icon library package for React Native applications",
  "version": "0.131.0",
  "license": "ISC",
  "homepage": "https://lucide.dev",
  "bugs": "https://github.com/lucide-icons/lucide/issues",
  "repository": {
    "type": "git",
    "url": "https://github.com/lucide-icons/lucide.git",
    "directory": "packages/lucide-react-native"
  },
  "author": "Eric Fennis",
  "amdName": "lucide-react-native",
  "main": "dist/cjs/lucide-react-native.js",
  "main:umd": "dist/umd/lucide-react-native.js",
  "module": "dist/esm/lucide-react-native.js",
  "unpkg": "dist/umd/lucide-react-native.min.js",
  "typings": "dist/lucide-react-native.d.ts",
  "react-native": "dist/esm/lucide-react-native.js",
  "sideEffects": false,
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "pnpm clean && pnpm copy:license && pnpm build:icons && pnpm build:bundles && pnpm build:types",
    "copy:license": "cp ../../LICENSE ./LICENSE",
    "clean": "rm -rf dist && rm -rf stats && rm -rf ./src/icons/*.js",
    "build:icons": "build-icons --output=./src --templateSrc=./scripts/exportTemplate.mjs --renderUniqueKey --iconFileExtension=.ts --exportFileName=index.ts --withAliases --aliasesFileExtension=.ts",
    "build:types": "node ./scripts/buildTypes.mjs",
    "build:bundles": "rollup -c ./rollup.config.mjs",
    "test": "vitest run",
    "version": "pnpm version --git-tag-version=false"
  },
  "devDependencies": {
    "@lucide/rollup-plugins": "workspace:*",
    "@lucide/build-icons": "workspace:*",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@types/prop-types": "^15.7.5",
    "@types/react": "^18.0.21",
    "@vitejs/plugin-react": "^2.1.0",
    "prop-types": "^15.7.2",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-native": "^0.69.0",
    "react-native-svg": "^13.0.0",
    "rollup": "^3.5.1",
    "typescript": "^4.8.4",
    "vite": "^3.1.0",
    "vitest": "^0.24.3"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^16.5.1 || ^17.0.0 || ^18.0.0",
    "react-native": "*",
    "react-native-svg": "^12.0.0 || ^13.0.0"
  }
}
