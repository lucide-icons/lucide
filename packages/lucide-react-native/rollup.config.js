import plugins from '../../rollup.plugins';
import pkg from './package.json';

const packageName = 'LucideReact';
const outputFileName = 'lucide-react-native';
const rootDir = 'packages/lucide-react-native'; // It runs from the root
const outputDir = `${rootDir}/dist`;
const inputs = [`${rootDir}/src/lucide-react-native.js`];
const bundles = [
  {
    format: 'umd',
    inputs,
    outputDir,
    minify: true,
  },
  {
    format: 'umd',
    inputs,
    outputDir,
  },
  {
    format: 'cjs',
    inputs,
    outputDir,
    preserveModules: true,
  },
  {
    format: 'esm',
    inputs,
    outputDir,
    preserveModules: true,
  },
];

const configs = bundles
  .map(({ inputs, outputDir, format, minify, preserveModules }) =>
    inputs.map(input => ({
      input,
      plugins: plugins(pkg, minify),
      external: ['react', 'prop-types', 'lucide', 'react-native-svg'],
      output: {
        name: packageName,
        ...(preserveModules
          ? {
              dir: `${outputDir}/${format}`,
            }
          : {
              file: `${outputDir}/${format}/${outputFileName}${minify ? '.min' : ''}.js`,
            }),
        format,
        preserveModules,
        sourcemap: true,
        globals: {
          react: 'react',
          'react-native-svg': 'react-native-svg',
          'prop-types': 'PropTypes',
          lucide: 'lucide',
        },
      },
    })),
  )
  .flat();

export default configs;
