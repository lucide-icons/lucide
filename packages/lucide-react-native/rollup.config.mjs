import plugins from '@lucide/rollup-plugins';
import pkg from './package.json' assert { type: 'json' };

const packageName = 'LucideReact';
const outputFileName = 'lucide-react-native';
const outputDir = 'dist';
const inputs = ['src/lucide-react-native.ts'];
const bundles = [
  {
    format: 'umd',
    inputs,
    outputDir,
    minify: true,
  },
  {
    format: 'umd',
    inputs,
    outputDir,
  },
  {
    format: 'cjs',
    inputs,
    outputDir,
    preserveModules: true,
  },
  {
    format: 'es',
    inputs,
    outputDir,
  },
  {
    format: 'esm',
    inputs,
    outputDir,
    preserveModules: true,
  },
];

const configs = bundles
  .map(({ inputs, outputDir, format, minify, preserveModules }) =>
    inputs.map(input => ({
      input,
      plugins: plugins(pkg, minify),
      external: ['react', 'prop-types', 'react-native-svg'],
      output: {
        name: packageName,
        ...(preserveModules
          ? {
              dir: `${outputDir}/${format}`,
              exports: 'auto',
            }
          : {
              file: `${outputDir}/${format}/${outputFileName}${minify ? '.min' : ''}.js`,
            }),
        format,
        preserveModules,
        sourcemap: true,
        globals: {
          react: 'react',
          'react-native-svg': 'react-native-svg',
          'prop-types': 'PropTypes',
        },
      },
    })),
  )
  .flat();

export default configs;
