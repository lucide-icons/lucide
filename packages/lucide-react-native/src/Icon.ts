import { createElement, forwardRef, type FunctionComponent } from 'react';
import * as NativeSvg from 'react-native-svg';
import defaultAttributes, { childDefaultAttributes } from './defaultAttributes';
import { IconNode, LucideProps } from './types';

interface IconComponentProps extends LucideProps {
  iconNode: IconNode;
}

/**
 * Lucide icon component
 *
 * @component Icon
 * @param {object} props
 * @param {string} props.color - The color of the icon
 * @param {number} props.size - The size of the icon
 * @param {number} props.strokeWidth - The stroke width of the icon
 * @param {boolean} props.absoluteStrokeWidth - Whether to use absolute stroke width
 * @param {string} props.className - The class name of the icon
 * @param {IconNode} props.children - The children of the icon
 * @param {IconNode} props.iconNode - The icon node of the icon
 *
 * @returns {ForwardRefExoticComponent} LucideIcon
 */
const Icon = forwardRef<SVGSVGElement, IconComponentProps>(
  (
    {
      color = 'currentColor',
      size = 24,
      strokeWidth = 2,
      absoluteStrokeWidth,
      children,
      iconNode,
      ...rest
    },
    ref,
  ) => {
    const customAttrs = {
      stroke: color,
      strokeWidth: absoluteStrokeWidth ? (Number(strokeWidth) * 24) / Number(size) : strokeWidth,
      ...rest,
    };

    return createElement(
      NativeSvg.Svg as unknown as string,
      {
        ref,
        ...defaultAttributes,
        width: size,
        height: size,
        ...customAttrs,
      },
      [
        ...iconNode.map(([tag, attrs]) => {
          const upperCasedTag = (tag.charAt(0).toUpperCase() +
            tag.slice(1)) as keyof typeof NativeSvg;
          // duplicating the attributes here because generating the OTA update bundles don't inherit the SVG properties from parent (codepush, expo-updates)
          return createElement(
            NativeSvg[upperCasedTag] as FunctionComponent<LucideProps>,
            { ...childDefaultAttributes, ...customAttrs, ...attrs } as LucideProps,
          );
        }),
        ...((Array.isArray(children) ? children : [children]) || []),
      ],
    );
  },
);

export default Icon;
