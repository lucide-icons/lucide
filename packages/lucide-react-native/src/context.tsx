import { createContext, type ReactNode, useContext } from 'react';

const LucideContext = createContext<{
  size?: number;
  fill?: string;
  color?: string;
  strokeWidth?: number;
  absoluteStrokeWidth?: boolean;
}>({
  size: 24,
  fill: 'none',
  color: 'currentColor',
  strokeWidth: 2,
  absoluteStrokeWidth: false,
});

interface LucideProviderProps {
  children: ReactNode;
  size?: number;
  fill?: string;
  color?: string;
  strokeWidth?: number;
  absoluteStrokeWidth?: boolean;
}

export function LucideProvider({ children, ...props }: LucideProviderProps) {
  return <LucideContext.Provider value={props}>{children}</LucideContext.Provider>;
}

export const useLucideContext = () => useContext(LucideContext);
