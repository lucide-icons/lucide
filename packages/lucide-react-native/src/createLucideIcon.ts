import { forwardRef, createElement, ReactSVG, ReactNode, FunctionComponent } from 'react';
import PropTypes from 'prop-types';
import * as NativeSvg from 'react-native-svg';
import defaultAttributes from './defaultAttributes';
import type { SvgProps } from 'react-native-svg';
type IconNode = [elementName: keyof ReactSVG, attrs: Record<string, string>][]


export interface LucideProps extends SvgProps {
  size?: string | number
}

const createLucideIcon = (iconName: string, iconNode: IconNode) => {
  const Component = forwardRef(
    ({ color = 'currentColor', size = 24, strokeWidth = 2, children, ...rest }: LucideProps, ref) =>
      createElement(
        NativeSvg.Svg,
        {
          ref,
          ...defaultAttributes,
          width: size,
          height: size,
          stroke: color,
          strokeWidth,
          ...rest,
        },
        [
          ...iconNode.map(([tag, attrs]) => {
            const upperCasedTag = (tag.charAt(0).toUpperCase() + tag.slice(1)) as keyof (typeof NativeSvg);
            return createElement(NativeSvg[upperCasedTag] as FunctionComponent<Record<string, string>>, attrs);
          }),
          ...(children as ReactNode[] || []),
        ],
      ),
  );

  Component.propTypes = {
    color: PropTypes.string,
    size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    strokeWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  };

  Component.displayName = `${iconName}`;

  return Component;
};

export default createLucideIcon;
