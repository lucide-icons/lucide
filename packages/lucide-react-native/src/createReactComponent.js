import { forwardRef, createElement } from 'react';
import PropTypes from 'prop-types';
import * as NativeSvg from 'react-native-svg';
import defaultAttributes from './defaultAttributes';

const createReactComponent = (iconName, iconNode) => {
  const Component = forwardRef(
    ({ color = 'currentColor', size = 24, strokeWidth = 2, children, ...rest }, ref) =>
      createElement(
        NativeSvg.Svg,
        {
          ref,
          ...defaultAttributes,
          width: size,
          height: size,
          stroke: color,
          strokeWidth,
          ...rest,
        },
        [
          ...iconNode.map(([tag, attrs]) => {
            const uppercasedTag = tag.charAt(0).toUpperCase() + tag.slice(1);
            return createElement(NativeSvg[uppercasedTag], attrs);
          }),
          ...(children || []),
        ],
      ),
  );

  Component.propTypes = {
    color: PropTypes.string,
    size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    strokeWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  };

  Component.displayName = `${iconName}`;

  return Component;
};

export default createReactComponent;
