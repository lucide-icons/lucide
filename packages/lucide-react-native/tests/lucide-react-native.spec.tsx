import { describe, it, expect, vi, beforeEach } from 'vitest';
import { cleanup, render } from '@testing-library/react'
import { Edit2, Grid, Pen } from '../src/lucide-react-native'

vi.mock('react-native-svg')

type Attributes = Record<string, { value: unknown}>

describe('Using lucide icon components', () => {
  it('should render an component', () => {
    const { container } = render(<Grid /> );

    expect( container.innerHTML ).toMatchSnapshot();
  });

  it('should adjust the size, stroke color and stroke width', () => {
    const testId = 'grid-icon';
    const { container, getByTestId } = render(
      <Grid
        data-testid={testId}
        size={48}
        stroke="red"
        strokeWidth={4}
      />,
    );

    const { attributes } = getByTestId(testId);
    expect((attributes as unknown as Attributes).stroke.value).toBe('red');
    expect((attributes as unknown as Attributes).width.value).toBe('48');
    expect((attributes as unknown as Attributes).height.value).toBe('48');
    expect((attributes as unknown as Attributes)['stroke-width'].value).toBe('4');

    expect( container.innerHTML ).toMatchSnapshot();
  });

  it('should render the alias icon', () => {
    const testId = 'pen-icon';
    const { container } = render(
      <Pen
        data-testid={testId}
        size={48}
        stroke="red"
        strokeWidth={4}
      />,
    );

    const PenIconRenderedHTML = container.innerHTML

    cleanup()

    const { container: Edit2Container } = render(
      <Edit2
        data-testid={testId}
        size={48}
        stroke="red"
        strokeWidth={4}
      />,
    );

    expect(PenIconRenderedHTML).toBe(Edit2Container.innerHTML)
  });


  it('should not scale the strokeWidth when absoluteStrokeWidth is set', () => {
    const testId = 'grid-icon';
    const { container, getByTestId } = render(
      <Grid
        data-testid={testId}
        size={48}
        stroke="red"
        absoluteStrokeWidth
      />,
    );

    const { attributes } = getByTestId(testId) as unknown as{ attributes: Record<string, { value: string }>};
    expect(attributes.stroke.value).toBe('red');
    expect(attributes.width.value).toBe('48');
    expect(attributes.height.value).toBe('48');
    expect(attributes['stroke-width'].value).toBe('1');

    expect( container.innerHTML ).toMatchSnapshot();
  });

  it('should work with a single child component', () => {
    const testId = "single-child";
    const childId = "child";

    const { container, getByTestId } = render(
      <Grid data-testid={testId}>
        <Grid data-testid={childId}/>
      </Grid>
    );
    const { children} = getByTestId(testId) as unknown as{ children: HTMLCollection};
    const lastChild = children[children.length -1];

    expect(lastChild).toEqual(getByTestId(childId));
    expect(container.innerHTML).toMatchSnapshot();
  })

  it('should work with several children components', () => {
    const testId = "multiple-children";
    const childId1 = "child1";
    const childId2 = "child2";

    const { container, getByTestId } = render(
      <Grid data-testid={testId}>
        <Grid data-testid={childId1}/>
        <Grid data-testid={childId2}/>
      </Grid>
    );
    const {children} = getByTestId(testId) as unknown as{ children: HTMLCollection};
    const child1 = children[children.length - 2];
    const child2 = children[children.length - 1];

    expect(child1).toEqual(getByTestId(childId1));
    expect(child2).toEqual(getByTestId(childId2));
    expect(container.innerHTML).toMatchSnapshot();
  })
})
