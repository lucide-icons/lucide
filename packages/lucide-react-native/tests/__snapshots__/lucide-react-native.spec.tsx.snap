// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Using lucide icon components > should adjust the size, stroke color and stroke width 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="48"
     height="48"
     viewbox="0 0 24 24"
     fill="none"
     stroke="red"
     stroke-width="4"
     stroke-linecap="round"
     stroke-linejoin="round"
>
  <rect fill="none"
        stroke="red"
        stroke-width="4"
        stroke-linecap="round"
        stroke-linejoin="round"
        width="18"
        height="18"
        x="3"
        y="3"
        rx="2"
  >
  </rect>
  <path fill="none"
        stroke="red"
        stroke-width="4"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 9h18"
  >
  </path>
  <path fill="none"
        stroke="red"
        stroke-width="4"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 15h18"
  >
  </path>
  <path fill="none"
        stroke="red"
        stroke-width="4"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M9 3v18"
  >
  </path>
  <path fill="none"
        stroke="red"
        stroke-width="4"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M15 3v18"
  >
  </path>
</svg>
`;

exports[`Using lucide icon components > should duplicate properties to children components 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewbox="0 0 24 24"
     fill="red"
     stroke="white"
     stroke-width="10"
     stroke-linecap="round"
     stroke-linejoin="round"
     data-testid="multiple-children"
>
  <rect fill="red"
        stroke="white"
        stroke-width="10"
        stroke-linecap="round"
        stroke-linejoin="round"
        width="18"
        height="18"
        x="3"
        y="3"
        rx="2"
  >
  </rect>
  <path fill="red"
        stroke="white"
        stroke-width="10"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 9h18"
  >
  </path>
  <path fill="red"
        stroke="white"
        stroke-width="10"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 15h18"
  >
  </path>
  <path fill="red"
        stroke="white"
        stroke-width="10"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M9 3v18"
  >
  </path>
  <path fill="red"
        stroke="white"
        stroke-width="10"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M15 3v18"
  >
  </path>
</svg>
`;

exports[`Using lucide icon components > should not scale the strokeWidth when absoluteStrokeWidth is set 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="48"
     height="48"
     viewbox="0 0 24 24"
     fill="none"
     stroke="red"
     stroke-width="1"
     stroke-linecap="round"
     stroke-linejoin="round"
>
  <rect fill="none"
        stroke="red"
        stroke-width="1"
        stroke-linecap="round"
        stroke-linejoin="round"
        width="18"
        height="18"
        x="3"
        y="3"
        rx="2"
  >
  </rect>
  <path fill="none"
        stroke="red"
        stroke-width="1"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 9h18"
  >
  </path>
  <path fill="none"
        stroke="red"
        stroke-width="1"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 15h18"
  >
  </path>
  <path fill="none"
        stroke="red"
        stroke-width="1"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M9 3v18"
  >
  </path>
  <path fill="none"
        stroke="red"
        stroke-width="1"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M15 3v18"
  >
  </path>
</svg>
`;

exports[`Using lucide icon components > should render an component 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewbox="0 0 24 24"
     fill="none"
     stroke="currentColor"
     stroke-width="2"
     stroke-linecap="round"
     stroke-linejoin="round"
>
  <rect fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        width="18"
        height="18"
        x="3"
        y="3"
        rx="2"
  >
  </rect>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 9h18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 15h18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M9 3v18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M15 3v18"
  >
  </path>
</svg>
`;

exports[`Using lucide icon components > should work with a single child component 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewbox="0 0 24 24"
     fill="none"
     stroke="currentColor"
     stroke-width="2"
     stroke-linecap="round"
     stroke-linejoin="round"
     data-testid="single-child"
>
  <rect fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        width="18"
        height="18"
        x="3"
        y="3"
        rx="2"
  >
  </rect>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 9h18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 15h18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M9 3v18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M15 3v18"
  >
  </path>
  <svg xmlns="http://www.w3.org/2000/svg"
       width="24"
       height="24"
       viewbox="0 0 24 24"
       fill="none"
       stroke="currentColor"
       stroke-width="2"
       stroke-linecap="round"
       stroke-linejoin="round"
       data-testid="child"
  >
    <rect fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          width="18"
          height="18"
          x="3"
          y="3"
          rx="2"
    >
    </rect>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M3 9h18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M3 15h18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M9 3v18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M15 3v18"
    >
    </path>
  </svg>
</svg>
`;

exports[`Using lucide icon components > should work with several children components 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewbox="0 0 24 24"
     fill="none"
     stroke="currentColor"
     stroke-width="2"
     stroke-linecap="round"
     stroke-linejoin="round"
     data-testid="multiple-children"
>
  <rect fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        width="18"
        height="18"
        x="3"
        y="3"
        rx="2"
  >
  </rect>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 9h18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M3 15h18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M9 3v18"
  >
  </path>
  <path fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        d="M15 3v18"
  >
  </path>
  <svg xmlns="http://www.w3.org/2000/svg"
       width="24"
       height="24"
       viewbox="0 0 24 24"
       fill="none"
       stroke="currentColor"
       stroke-width="2"
       stroke-linecap="round"
       stroke-linejoin="round"
       data-testid="child1"
  >
    <rect fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          width="18"
          height="18"
          x="3"
          y="3"
          rx="2"
    >
    </rect>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M3 9h18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M3 15h18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M9 3v18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M15 3v18"
    >
    </path>
  </svg>
  <svg xmlns="http://www.w3.org/2000/svg"
       width="24"
       height="24"
       viewbox="0 0 24 24"
       fill="none"
       stroke="currentColor"
       stroke-width="2"
       stroke-linecap="round"
       stroke-linejoin="round"
       data-testid="child2"
  >
    <rect fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          width="18"
          height="18"
          x="3"
          y="3"
          rx="2"
    >
    </rect>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M3 9h18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M3 15h18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M9 3v18"
    >
    </path>
    <path fill="none"
          stroke="currentColor"
          stroke-width="2"
          stroke-linecap="round"
          stroke-linejoin="round"
          d="M15 3v18"
    >
    </path>
  </svg>
</svg>
`;
